components/sidebar/sidebar_p3a.cc

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/sidebar/sidebar_p3a.h"
+
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/sidebar/pref_names.h"
+#include "brave/components/sidebar/sidebar_service.h"
+#include "components/prefs/pref_service.h"
+
+namespace sidebar {
+namespace p3a {
+
+const char kSidebarEnabledHistogramName[] = "Brave.Sidebar.Enabled";
+
+SidebarP3A::SidebarP3A(PrefService* profile_prefs)
+    : profile_prefs_(profile_prefs) {
+  profile_pref_change_registrar_.Init(profile_prefs);
+  profile_pref_change_registrar_.Add(
+      kSidebarShowOption,
+      base::BindRepeating(&SidebarP3A::RecordEnabledSetting, AsWeakPtr()));
```

Comment:
you don't need a weak pointer for this, `base::Unretained` can be safely used here